src/wp-includes/plugin.php

```diff
@@ -39,6 +39,38 @@
 if ( ! isset( $wp_current_filter ) )
 	$wp_current_filter = array();
 
+/**
+ * Check that the callback parameters match the number passed to add_action() or add_filter().
+ *
+ * @since 1.3.0
+ *
+ * @param $function_name   add_action() or add_filter(), as returned by __FUNCTION__.
+ * @param $function_to_add The callback to check.
+ * @param $accepted_args   The number of arguments claimed.
+ */
+function _check_callback_params($function_name, $function_to_add, $accepted_args)
+{
+    if ( defined( 'CP_STRICT' ) && CP_STRICT ) {
+        if ( is_string( $function_to_add ) && function_exists( $function_to_add ) ) {
+            $rf = new ReflectionFunction( $function_to_add );
+            $params = $rf->getParameters();
+            $numParams = count( $params );
+            if ( $numParams == $accepted_args ) {
+                // we're all good
+            } elseif ( $params < $accepted_args ) {
```

Comment:
If there's no early bail or nothing to do in the first true statement here could we make the 2nd one the only defined condition?

Reply:
We could, but I was thinking we might want to log success later on?

Reply:
Yeah that makes sense :+1:

Reply:
This should be added if/when we add the code to log success, not as a placeholder.